Archive/Draft/RFCxxxx-PS7-Bootstrapper.md

```diff
@@ -0,0 +1,256 @@
+with regards to resource management which results in not only leaks but security issues.
+Resulting Rust binary code is very similar in size and performance as C/C++.
+Rust built code is supported in Windows.
+
+Rust uses the MS C runtime at execution, but we should statically link so that there aren't external dependencies that may be missing on the system.
+
+Need to evaluate what APIs are available on the smallest Windows SKU for downloading.
+
+This is the current preferred option.
+
+### Installer
+
+MSI has dependencies that are not available on Windows SKUs so would not be appropriate as the installer for the bootstrapper.
+Instead, we should look at using [Squirrel](https://github.com/Squirrel/Squirrel.Windows) which supports installing even if the product (existing version)
+is already in use.
+
+The bootstrapper would download the installer using Windows Update as the store to avoid issues with enterprises not allowing
+network access to other sites.
+The installer and binaries may need to be Windows signed instead of Microsoft signed.
+Packages of PS7 outside of this installer used by the boostrapper would continue to be Microsoft signed.
+
+### User Experience
+
+Since the initial bootstrapping will take time to download and install, the bootstrapper should have progress information and
+indicator during download and installation via stderr.
+Stderr is used instead of stdout so that user redirection of stdout only contains output from successful execution of their commandline.
+Use of stderr for informational messages is a convention used by native commands already.
+
+Should the install or download fail, the bootstrapper needs to output appropriate information and direct user to documentation on alternate
+install method and return a non-zero exit code.
+
+### Auto-updating
+
+Upon startup of the bootstrapper (following rules of [admin vs non-admin](#admin-vs-non-admins)) will detect if the installed
+version is installed by the bootstrapper and if so, if it is the latest stable version.
+
+If the version is older, then it will asynchronously download updated PS7 and continue with execution.
+A message will be emitted that on next startup PS7 will be updated.
+Upon next startup, if a newer version is downloaded, then the signature will be validated and installed providing
+progress information to the user.
+
+In general, auto-updates will be to a servicing release like 7.x.y to 7.x.y+1.
+However, when 7.x is out of support, then it should auto-update to 7.x+1.
+This is consistent with how our Microsoft Update support works to keep users patched and on a supported release.
+
+The auto-update applies regardless if PS7 is started interactively or via automation (explicitly using the `-noninteractive`, `-file`, or `-command` switch).
+
+### Configuration
+
+Some registry configuration (settable via Group Policy) would be needed to disable auto-updating.
```

Comment:
This is another reason for doing updates through WU, it's something already understood by admins and offers the ability for orgs to cache the update files on their own servers for distribution rather than having to find another way to get that data from the net.